### ISSUE TITLE
WM run under nvidia-xrun refuses to terminate

### ISSUE THREAD

**Issue by Uldreth** [Witko/nvidia-xrun#31]
Hello,

I have used nvidia-xrun on my Antergos system before, without issues. Then I switched to a different distro, didn't use it anymore, then switched back to Antergos. Only a span of about one or two weeks went by between the two.

In the current Antergos installation I use the LTS kernel and accordingly, the nvidia-lts drivers. **I don't know if this is an issue or not**, in the sense that the GPU *does* seem to be working (so question: do I need to change something in nvidia-xorg.conf to accomodate the lts driver?).

My first and foremost issue is that when I use nvidia-xrun to launch a window manager (currently a full XFCE desktop, but I also tried openbox before) via appending `exec startxfce4 ` to the nvidia-xinitrc file you have included (which have been moved to be ~/.nvidia-xinitrc), if I try to use the session logouter, or manually log out via terminal, instead of dropping to tty, it starts exiting the WM/X and then it **restarts**.

I am certain that even if it terminates the actual window manager, it doesn't terminate X, because your script had been set up to unload the nvidia kernel modules and turn off the GPU if the xinit command finishes running. This doesn't happen. The only way I managed to cease the session is to `pkill X`, which also kills my other X session. If I `pkill X`, then your script *does* unload the modules/turn off the GPU.

Although I am a bash newb, I did skim your script and I see absolutely nothing that could cause such a behaviour and I am totally dumbfounded.

--------------------------------------------

I have some other issues too, which I am not sure if it is related or not. My keyboard layout is set to hungarian, rather than english, and both my usual X session (via my usual .xinitrc) and the tty recognizes. The session launched by nvidia-xrun doesn't. I have tried to append the contents of my /etc/X11/xorg.conf.d/00-keyboard.conf file into your nvidia-xorg.conf, but nothing happened.

On top of that, running firefox from the nvidia-xrun session results in firefox launching, but not reacting to me trying to load any sites. I do have wifi connection, so this is not a connection issue, it is *not even trying* to load sites. I did not have time to test with another browser so far or test for consistency.

*Further remarks:* 

* The brower issue is seemingly solved when the window manager restart itself (via my primary issue), but I did not have time to test for consistency.

* Bumblebee itself is behaving weirdly. There are some things that work via `optirun`, but not `primusrun`, however glxspheres64 render via the dGPU for both commands. In particular, Steam, both Windows Steam via wine and Linux Steam natively, fail to start games with optirun/primusrun enabled, even if Steam is launched via optirun/primusrun OR the game is launched via the `primusrun %command% ` modifier. In fact, I `primusrun startx` once and my programs started with primusrun, but Steam didn't. I am only mentioning this, in case it reveals a deeper issue

**Comment by Witko** [comment]
Hi @Uldreth ,
its hard to judge what might be the issue. I'd say closer cooperation would be required as i dont have possibilities to try with other distros. Does it behave the same with all window managers? I switched to plasma session as i can have there everything and it works nicely. For the price of some Ram of course.

**Comment by Uldreth** [comment]
@Witko 

I have used it with openbox and xfce4, both had the same issues. I can try checking other WMs as well. Note that although for my *main* X session, I use i3-gaps, I did use openbox for a while and I had no issues.

Maybe I could also change .nvidia-xinitrc to be like my own xinitrc except the two xrandr lines added. I will get back to you with the results when I got the time, probably later today (depending on where you live it might not be today for you).

You do use Arch Linux right? If so, there should be no essential difference, Antergos is basically Arch with an installer and some custom repos with themes added. Also I installed nvidia-xrun from the AUR.

**Comment by Witko** [comment]
Hi @Uldreth ,
its hard to judge what might be the issue. I'd say closer cooperation would be required as i dont have possibilities to try with other distros. Does it behave the same with all window managers? I switched to plasma session as i can have there everything and it works nicely. For the price of some Ram of course.

**Comment by Witko** [comment]
Hi @Uldreth ,
i just realized i answered the same thing twice - not sure how:) Is this still an issue?

**Comment by Uldreth** [comment]
Hi @Witko , sorry for the late response. Truth to be told, I abandoned nvidia-xrun after I had this issue. With that said, later on when I tried to use bumblebee on Arch, I encountered some significant issues with bbswitch. Usually the dGPU didn't shut off, and I couldn't force it to shut off either via bbswitch. I tried to do some research and it seemed some systemd - related thing was causing the issue, unfortunately I don't remember what and there was no easy fix. This lead me to completely abandon bbswitch-based solutions (I currently run Solus, but on Arch I ended up using a self-made script that did a bbswitch-less, reboot-requiring switching :) ).

So unfortunately I cannot say if it is still problematic - aside from the fact that I find it likely my bbswitch issues would have caused problems with nvidia-xrun as well, since it relies on bbswitch.
